File: Creational/Abstract Factory/Python/ConcreteFactory2.py
from AbstractFactory import *
from ConcreteProductA_2 import *
from ConcreteProductB_2 import *
from ConcreteProductC_2 import *

class ConcreteFactory2(AbstractFactory):
	
	def create_productA(self):
		return ConcreteProductA_2()
		
	def create_productB(self):
		return ConcreteProductB_2()
		
	def create_productC(self):
		return ConcreteProductC_2()

File: Creational/Abstract Factory/Golang/ConcreteFactory2.go
package main

type ConcreteFactory2 struct{}

func NewConcreteFactory2() AbstractFactory {
	return &ConcreteFactory2{}
}

func (*ConcreteFactory2) CreateProductA() AbstractProductA {
	return &ConcreteProductA_2{}
}

func (*ConcreteFactory2) CreateProductB() AbstractProductB {
	return &ConcreteProductB_2{}
}

func (*ConcreteFactory2) CreateProductC() AbstractProductC {
	return &ConcreteProductC_2{}
}
